### ISSUE TITLE
no limitation on the inputs?

### ISSUE THREAD

**Issue by youngchannelforyou** [state-spaces/mamba#246]
Hello! First of all, thank you for creating such an amazing model. As a Korean, my English might be a bit awkward. From what I understood while training the model, I thought that Mamba can accept inputs without any limitations. However, when tokenizing text or performing operations within the model, is it necessary to align the inputs consistently with the data batch size? Is there really no limitation on the inputs?

**Comment by tridao** [comment]
I'm not sure I understand. The Mamba module input/output is similar to many sequence-to-sequence modules such as attention or MLP in Transformer, i.e. of shape (batch, seqlen, hidden_dim). The Mamba language model takes in input_ids of shape (batch, seqlen) just like other language models.

**Comment by zigzagcai** [comment]
As I understand it, mamba is similar to other seq2seq models. It learns the mapping relationship of (b, l, d)-> (b, l, d). 
So, within a batch or across multiple batches, the sequence length needs to be consistent.

**Comment by zigzagcai** [comment]
Of course, for inputs that are not consistent, we can use padding to make the seqlen consistent within a batch  or across multiple batches.
Moreover, we can fully exploit the characteristics of the seq2seq models so that inputs are no longer restricted. For example,

- Transformer: We can break down the attention calculation into basic 1row attention
- Mamba: We can reset the hidden states when multiple inputs are stitched together.
